### ISSUE TITLE
`List.hs`: решить `seqOptional` без `twiceOptional`

### ISSUE THREAD

**Issue by ulysses4ever** [MaryVtyurina/course#3]
После того, как разобрана голова списка, надо разбирать результат рекурсивного вызова.

**Comment by MaryVtyurina** [comment]
А результат рекурсивного вызова разбирать как голову на два случая?

**Comment by ulysses4ever** [comment]
Нет. Поскольку у вас тип результата Optional, то Разбирать так же, как и голову, то есть Full / Empty.
